src/node/desktop/src/main/detect-r.ts

```diff
@@ -55,34 +58,56 @@ function executeCommand(command: string): Expected<string> {
   }
 }
 
-export async function promptUserForR(): Promise<Expected<string | null>> {
-  // nothing to do if RSTUDIO_WHICH_R is set
-  const rstudioWhichR = getenv('RSTUDIO_WHICH_R');
-  if (rstudioWhichR) {
-    return ok(rstudioWhichR);
-  }
+export async function promptUserForR(platform = process.platform): Promise<Expected<string | null>> {
+  if (platform === 'win32') {
+    const desktop = await import('../native/desktop.node');
 
-  // discover available R installations
-  const rInstalls = findRInstallationsWin32();
-  if (rInstalls.length === 0) {
-    return err();
-  }
+    const storeConfig = new Store({ defaults: kDesktopOptionDefaults });
```

Comment:
@matheustavaresdev I just merged changes with how preferences work. I ended up renaming `DesktopOptions` to `ElectronDesktopOptions` so it is differentiated from the `legacyPreferenceManager` which required us supporting the platform-specific code for the preference store.

So I think all that needs to be done is updating `ElectronDesktopOptions` to set and retrieve this new option. Since it didn't exist in Qt, `legacyPreferenceManager` has no need to implement it.

Reply:
Fixed on https://github.com/rstudio/rstudio/pull/10749/commits/67a626fc3a8fcd1a1817d44d846e4a1b8fa81dbc